metrics.md

```diff
@@ -30,6 +30,55 @@ The following are the minimal metrics agreed to be conformed by the various clie
+| `beacon_data_availability_reconstructed_columns_total`            | Counter   | Total count of reconstructed columns                                      | On data column kzg verification  |
+| `beacon_data_availability_reconstruction_time_seconds`            | Histogram | Time taken to reconstruct columns                                      | On data column kzg verification  |
+| `beacon_data_column_sidecar_computation_seconds`                  | Histogram | Time taken to compute data column sidecar, including cells, proofs and inclusion proof                |  On data column sidecar computation            |
+| `beacon_data_column_sidecar_inclusion_proof_verification_seconds` | Histogram | Time taken to verify data column sidecar inclusion proof                          |  On data column sidecar inclusion proof verification  |
+| `beacon_kzg_verification_data_column_single_seconds`              | Histogram | Runtime of single data column kzg verification                                 | On single data column kzg verification  |
+| `beacon_kzg_verification_data_column_batch_seconds`               | Histogram | Runtime of batched data column kzg verification                                 | On batched data column kzg verification |
+| `beacon_custody_columns_count_total`                              | Counter     | Total count of columns in custody within the data availability boundary                                     | On custody collecting and verification |
+
+#### Gossipsub metrics
+Gossipsub domain metrics should contain topic `data_column_sidecar_{subnet_id}` as labels.
+
+| Name | Metric type | Usage | Sample collection event |
+|--------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `gossipsub_topic_msg_sent_counts_total` | Counter | Number of gossip messages sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_sent_bytes_total` | Counter | Number of bytes sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_recv_counts_unfiltered_total` | Counter | Number of gossip messages received from each topic (including duplicates)  | On receiving a message from a topic including duplicates|
+| `gossipsub_topic_msg_recv_bytes_unfiltered_total` | Counter | Number of bytes received from each topic (including duplicates) |  On receiving a message from a topic including duplicates |
+| `gossipsub_topic_msg_recv_counts_total` | Counter | Number of gossip messages received from each topic (deduplicated)  | On receiving a message from a topic deduplicated |
+| `gossipsub_topic_msg_recv_bytes_total` | Counter | Number of bytes received from each topic (deduplicated) | On receiving a message from a topic deduplicated |
```

Comment:
I think these generic metrics would benefit multi-client monitoring  - although it would be difficult for clients to change them if they already exist.

Most of these metrics already exist in `rust-libp2p`, I wonder if other clients have equivalent metric for these? If it's hard to standardise these, perhaps it would still be useful to find out the equivalent for each client, so we could make it possible to have a single multi-client dashboard without standardising the metric names?

@tersec @nalepae @zilm13 @matthewkeil

Reply:
btw the duplicate tracking metrics would be useful to analyse the effectiveness of `IDONTWANT`.

Reply:
We (Lodestar) are the owners of the `js-libp2p-gossipsub` repo and have updated those metric names to match the PR here.  That is still in draft though because this PR has not landed.

I do completely agree that unifying metric names across clients will be huge for both ethpandops and operators as a whole.  By making dashboards universal it will be trivial to switch to promote client diversity or to recover if a big bug is found in one.